source/tls_channel_handler.c

```diff
@@ -244,21 +238,19 @@ int aws_tls_ctx_options_init_client_mtls_with_pkcs11(
             aws_string_new_from_cursor(allocator, &pkcs11_options->private_key_object_label);
     }
 
-    struct aws_pkcs11_tls_op_handler *pkcs11_handler = aws_pkcs11_tls_op_handler_new(
+    struct aws_custom_key_op_handler *pkcs11_handler = aws_pkcs11_tls_op_handler_new(
```

Comment:
(commenting on lines above this) you can move all the validation and transforming of byte-cursors into strings over to the aws_pkcs11_tls_op_handler_new() function, instead of having it in tls_channel_handler.c